build.rs

```diff
@@ -0,0 +1,29 @@
+use std::process::Command;
+use std::error::Error;
+use std::env;
+use std::path::PathBuf;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let out_dir = env::var("OUT_DIR").unwrap();
+
+    let files: Vec<PathBuf> = std::fs::read_dir("references/java")?
+        .map(|p| p.ok().unwrap().path()).collect();
+
+    let output = Command::new("javac")
+        .args(files)
+        .arg("-d").arg(&out_dir).output()?;
+    if !(output.status.success()) {
+        panic!("Compilation of references/java failed ({}):\n{}", output.status.code().unwrap(), String::from_utf8(output.stderr)?);
+    }
+
+    let files: Vec<PathBuf> = std::fs::read_dir("references/scala")?
+        .map(|p| p.ok().unwrap().path()).collect();
+
+    let output = Command::new("scalac")
```

Comment:
this (and the javac dependency) are why i've committed the .class files in-tree directly, in fact. i want to be able to clone and hack on zvm without needing non-rust toolchains so classes used for regression (or otherwise) testing are perfectly fine to have in tree. at some point it might be worthwhile to move them to another repo, but as long as individual class files are <5kb or so i'm not sweating it

if `references/` grows past a meg i'll start getting worried.

that all said, i'm not a huge fan of arbitrary binary blobs being in-tree, so i'm committing source alongside them for the sake of reproducability. if i had CI set up, i'd like that to verify that committed source and class files under references line up.

Reply:
Reproducibility is a fair reason, so I'll drop that part of the PR. I do warn you that depending on javac/scalac is reasonable since compiling the classes is not within the scope of a JVM implementation itself (write once, run anywhere).